src/runtime/relax_vm/cuda/cuda_graph_builtin.cc

```diff
@@ -31,6 +31,46 @@ namespace tvm {
 namespace runtime {
 namespace relax_vm {
 
+struct CUDAGraphCaptureKey {
+  int64_t index;
+  ShapeTuple shape_expr;
+
+  CUDAGraphCaptureKey(int64_t index, const Optional<ShapeTuple>& shape_expr) : index(index) {
+    if (shape_expr) {
+      this->shape_expr = shape_expr.value();
+    }
+  }
+
+  bool operator==(const CUDAGraphCaptureKey& other) const {
```

Comment:
Use explicit structural CudaCaptureKeyHash and CudaCaptureKeyEqual

Reply:
reference https://github.com/apache/tvm/blob/main/include/tvm/node/structural_equal.h#L104